### PR TITLE
Added a cpnews service that functions as a syndication service

### PR DESCRIPTION
**Pull request by sturufous**
Implemented the ingestion of CP News articles as a separate service. This service ingests the Command News RSS 0.91 feed which provides an XML document with bare-bones items that contain only title and link tags. The article text has to be retrieved using a separate HTTP request to the link provided. As there are no created or published dates per item this may affect the detection of duplicate/updated entries (please advise). If this is an issue I can modify the code to extract the date/time from the article. I'd like to focus on the remote video editing infrastructure and tomorrow's demo for the rest of today if possible.

### CODE REVIEW

File: services/net/cpnews/CPNewsAction.cs [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    {
+        var response = await _httpClient.GetAsync(url);
+        var data = await response.Content.ReadAsStringAsync();
+
+        try
+        {
+            var xmlr = XmlReader.Create(new StringReader(data));
+            return SyndicationFeed.Load(xmlr);
+
+            // var feed = SyndicationFeed.Load(xmlr);
```
Comment (Fosol) [inline services/net/cpnews/CPNewsAction.cs:218]: Should be able to remove this comment.

File: services/net/cpnews/CPNewsAction.cs [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    }
+
+    /// <summary>
+    /// Extract the article text from the HTML document articleContent and remove all tags. This produces a rather
+    /// monolithic block of text. This method can be modified to refine the appearance of the output once we have
+    /// a subscriber interface to work with.
+    /// </summary>
+    /// <param name="articleContent">HTML encoded news article</param>
+    /// <returns>Article text only with tags removed</returns>
+    private static string SanitizeContent(string articleContent)
```
Comment (Fosol) [inline services/net/cpnews/CPNewsAction.cs:260]: Love to move this to an extension method in our `TNO.Core` library.

File: services/net/cpnews/CPNewsDataSourceManager.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+using Microsoft.Extensions.Options;
+using System.Text.Json;
+using TNO.API.Areas.Services.Models.DataSource;
+using TNO.Services.Actions.Managers;
+using TNO.Services.CPNews.Config;
+
+namespace TNO.Services.CPNews;
+
+/// <summary>
+/// CPNewsDataSourceManager class, provides a way to manage the cobews ingestion process for this data source.
```
Comment (Fosol) [inline services/net/cpnews/CPNewsDataSourceManager.cs:10]: spelling `cobews`

File: services/net/cpnews/Xml/Atom/AtomFeed.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+using System.Xml.Linq;
+using System.Xml.Serialization;
+
+namespace TNO.Services.CPNews.Xml;
+
+/// <summary>
+/// AtomFeed class, provides an object to read and write Atom syndication feed.
+/// </summary>
+[XmlRoot("atom")]
+public class AtomFeed
```
Comment (Fosol) [inline services/net/cpnews/Xml/Atom/AtomFeed.cs:12]: The duplication of the Xml folder isn't good.  Need to centralize this in the Services library.

File: services/net/cpnews/appsettings.Development.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  "Service": {
+    "MaxFailLimit": 5,
+    "ApiUrl": "http://host.docker.internal:40010/api",
+    "MediaTypes": ["CPNews"]
+  },
+  "Auth": {
+    "Keycloak": {
+      "Authority": "http://host.docker.internal:40001/auth/realms/tno",
+      "Audience": "tno-service-account",
+      "Client": "tno-service-account",
```
Comment (Fosol) [inline services/net/cpnews/appsettings.Development.json:17]: Checkout the latest changes in my prior PR for configuration.  There are some differences in the Keycloak section.

File: services/net/cpnews/appsettings.Staging.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      "Microsoft.AspNetCore": "Warning"
+    }
+  },
+  "Service": {
+    "MaxFailLimit": 5,
+    "ApiUrl": "http://api:8080",
+    "MediaTypes": ["CPNews"]
+  },
+  "Auth": {
+    "Keycloak": {
```
Comment (Fosol) [inline services/net/cpnews/appsettings.Staging.json:14]: Checkout the latest changes in my prior PR for configuration. There are some differences in the Keycloak section.

File: services/net/cpnews/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "AllowedHosts": "*",
+  "Service": {
+    "MaxFailLimit": 5,
+    "ApiUrl": "http://api:8080",
+    "MediaTypes": ["CPNews"],
+    "TimeZone": "Pacific Standard Time"
+  },
+  "Auth": {
+    "Keycloak": {
```
Comment (Fosol) [inline services/net/cpnews/appsettings.json:17]: Checkout the latest changes in my prior PR for configuration. There are some differences in the Keycloak section.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: services/net/cpnews/appsettings.Development.json (hunk 5 above)
Comment (Fosol) [reply services/net/cpnews/appsettings.Development.json:17]: Just remove the "Client" property

File: services/net/cpnews/appsettings.Staging.json (hunk 6 above)
Comment (Fosol) [reply services/net/cpnews/appsettings.Staging.json:14]: Just remove the "Client" property

File: services/net/cpnews/CPNewsDataSourceManager.cs (hunk 3 above)
Comment (sturufous) [reply services/net/cpnews/CPNewsDataSourceManager.cs:10]: Fixed

File: services/net/cpnews/appsettings.Development.json (hunk 5 above)
Comment (sturufous) [reply services/net/cpnews/appsettings.Development.json:17]: Removed it.

File: services/net/cpnews/appsettings.Staging.json (hunk 6 above)
Comment (sturufous) [reply services/net/cpnews/appsettings.Staging.json:14]: Removed it.

File: services/net/cpnews/CPNewsAction.cs (hunk 2 above)
Comment (sturufous) [reply services/net/cpnews/CPNewsAction.cs:260]: As this is a string processing method should it go to:

tno\libs\net\core\Extensions\StringExtensions.cs

?

File: services/net/cpnews/Xml/Atom/AtomFeed.cs (hunk 4 above)
Comment (Fosol) [reply services/net/cpnews/Xml/Atom/AtomFeed.cs:12]: yes

File: services/net/cpnews/CPNewsAction.cs (hunk 2 above)
Comment (Fosol) [reply services/net/cpnews/CPNewsAction.cs:260]: yes

**Review by Fosol — CHANGES_REQUESTED** [review]
Delete anything related to Syndication logic in this PR.  We already have a syndication feed service.  This service is for custom XML.

**Pull request closed without merging**